packages/cadl-ranch-specs/http/type/property/value-types/main.tsp

```diff
@@ -269,3 +269,19 @@ model UnionFloatLiteralProperty is ModelTemplate<43.125 | 46.875>;
 @route("/union/float/literal")
 @operationGroup
 interface UnionFloatLiteral extends ModelOperations<UnionFloatLiteralProperty, "46.875"> {}
+
+union BaseEnum {
+  string,
+  EnumValue1: "value1",
+}
+
+union ExtendedEnum {
+  BaseEnum,
+  EnumValue2: "value2",
+}
```

Comment:
I use the extended enum, as dev could write `property: BaseEnum.EnumValue1` as well.

Guess we first get the basic case covered.

Reply:
Seems weird. Why we don't use:
```
union BaseEnum {
  string,
  EnumValue1: "value1",
  EnumValue2: "value2",
}
```
directly?

Reply:
Dev may want to "extend" a Enum.

And it is already in use
https://github.com/Azure/azure-rest-api-specs/pull/27749#discussion_r1494047254

Reply:
Checked the logic for `getUnionAsEnum` and it support such defintion.

Reply:
Yeah, that is the "flatten" on `flattenedMembers`

Reply:
I've removed the hierarchy in this case, as the expected behavior is in discussion.